app/web_hub.go

```diff
@@ -433,6 +436,9 @@ func (h *Hub) Start() {
 					}
 				}
 			case msg := <-h.broadcast:
+				if metrics := h.app.Metrics(); metrics != nil {
+					metrics.DecrementWebSocketBroadcastHubElements(strconv.Itoa(h.connectionIndex), 1)
```

Comment:
I'm not sure if we should use a goroutine to call these functions. The prometheus client should do it asynchronously but this one is inside a select:

```go
h.app.Srv().Go(func() {
    metrics.DecrementWebSocketBroadcastHubElements(strconv.Itoa(h.connectionIndex), 1)
})
```

Reply:
There's no need to use a goroutine. The increment/decrement happens in-memory. The numbers are later scraped from the /metrics endpoint.

Reply:
So [in this case](https://github.com/mattermost/mattermost-server/blob/master/app/web_conn.go#L235-L237) is not needed

Reply:
Whoa boy .. yes indeed. Let's remove that. Would you be open to send a PR ?

Reply:
Sure! That's why I was confused about using goroutines